packages/flutter_tools/test/general.shard/commands/build_android_test.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:flutter_tools/src/android/android_sdk.dart';
+import 'package:flutter_tools/src/android/gradle.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/build_apk.dart';
+import 'package:flutter_tools/src/commands/build_appbundle.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+class CurrentDirectoryFlutterProjectFactory extends FlutterProjectFactory {
+  Directory redirectCurrentTo;
+
+  @override
+  FlutterProject fromDirectory(Directory directory) {
+    if (redirectCurrentTo != null && directory.path == fs.currentDirectory.path) {
+      return super.fromDirectory(redirectCurrentTo);
+    }
+    return super.fromDirectory(directory);
+  }
+}
+
+void main() {
+  Cache.disableLocking();
+
+  group('Build without full SDK installed', () {
+    AndroidSdk sdk;
+    Gradle gradle;
+    Directory tempDir;
+    CurrentDirectoryFlutterProjectFactory flutterProjectFactory;
+
+    final Map<Type, Generator> overrides = <Type, Generator>{
```

Comment:
nit: inline the overrides, so it's more clear what those are as this file grows in number of lines.

Reply:
Are you suggesting that I move this Map into each call site?

This will result in duplication, and besides, the parameters are defined in `setUp()` of the `group()` so they will be hanging around the top of the file anyway.

I don't have a strong opinion, so if what I said above is what you want me to do, I will do it.